### ISSUE TITLE
Edit database cell doesnt check or raise error when inputing wrong data type ? 

### ISSUE THREAD

**Issue by jimbasquiat** [sqlitebrowser/sqlitebrowser#2491]
Details for the issue
--------------------

#### What did you do?
In the edit database cell panel, trying to input 'hello' in the int primary key column.

#### What did you expect to see?
A warning saying i entered the wrong type

#### What did you see instead?
text is input without warning. No warning either when i click 'write changes' button
Additionaly, if i enter data in another column then the primary, I would expect the primary to automatically be given a number (from documentation, autoincrement is automatic in sqlite: https://sqlite.org/autoinc.html). This is also not the case.

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ ] Windows: ( _version:_ ___ )
- [x] Linux: ( _distro:ubuntu 18.04)
- [ ] macOS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [ ] 3.12.1
- [ ] 3.12.0
- [x] 3.11.2
- [ ] Other: ___

#### Did you also

- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by chrisjlocke** [comment]
SQLite doesn't care about the datatypes of a column.  You can happily put 'hello' in a numeric column.  This is by design of SQLite.  You can even put '12' (numerically, although I've put quotes around it... ignore that) in a text column.  The affinity (column type) is more about what data _should_ go in a cell and if SQLite can then store it neater.  If its meant to be numeric, it can try and change '24' to 24 to store it in an integer cell.  

With regard to your second point about "expect the primary to automatically be given a number" can you give explicit steps on what you're doing?

As an example, I have no data in this table.

![image](https://user-images.githubusercontent.com/7872218/100377856-04025500-300a-11eb-8623-c4f5034f5fb8.png)

I click the 'add record' button and immediately the primary key column has a value in it.

![image](https://user-images.githubusercontent.com/7872218/100377903-1e3c3300-300a-11eb-98b5-7054b83ac24b.png)

Now I suspect (as you're using an older version of DB4S) you might be entering data into the filter row.  But as you've not provided any information, that's solely a guess.  The newer version of DB4S hides this row if you have no data.

For the avoidance of doubt, if a column is set to be the primary key, you cannot enter textual data in this column.

![image](https://user-images.githubusercontent.com/7872218/100378093-64919200-300a-11eb-9181-bd6b8136c0b2.png)

**Comment by jimbasquiat** [comment]
Thanks a lot for making clearer the question about datatype.

Regarding the second question, yes i dont have the same behavior as you.

Consider the following table:

![image](https://user-images.githubusercontent.com/19773990/100380166-32822f00-300e-11eb-9a2d-25e50fd99d9c.png)

If i hit new record i will see the following:
![image](https://user-images.githubusercontent.com/19773990/100380213-50e82a80-300e-11eb-8b1d-58e6420721cf.png)

I would expect the column indicatorID to have the int 1 already. Additionaly it is possible for me to input text data here, which should also return an error like in your example. 
![image](https://user-images.githubusercontent.com/19773990/100380306-81c85f80-300e-11eb-9d62-298be24fbb82.png)


What am i doing wrong ?

**Comment by chrisjlocke** [comment]
I don't think that's valid SQLite - SQLite doesn't know what 'varchar(255)' is - it only understands text, numeric, etc.

Can you right-click on the 'indicators' table, select 'Copy create statement' and paste it here?

I'm guessing SQLite hasn't fully assigned that table a primary key.
Has that table got any meaningful data in it?

**Comment by jimbasquiat** [comment]
Thanks alot for your answer.

Sorry I m working on thet db now and just deleted that indicators table.. However here a similar table that is created (it works) with the varchar(255).

```
CREATE TABLE `search_terms` (
  `search_termID` int PRIMARY KEY,
  `symbolID` int,
  `topicID` int,
  `search_terms` varchar(255),
  `iteration` int,
  CONSTRAINT fk_symbolID
    FOREIGN KEY (symbolID)
    REFERENCES symbols(symbolID)
  CONSTRAINT fk_topicID
    FOREIGN KEY (topicID)
    REFERENCES topics(topicID)
)
```
and here what i see when clicking on modify table:

![image](https://user-images.githubusercontent.com/19773990/100382157-96a6f200-3012-11eb-8849-b5a378e5b790.png)

**Comment by chrisjlocke** [comment]
Yes, I guess you could call it a bug.  I'll raise it as one.
The code has spotted 'Primary Key' and slapped in the icon to denote that, but 'int' isn't a known type for SQLite, so it doesn't know what to do with it.  So, its wrong and DB4S isn't letting you know.  

Change the table to this:
`CREATE TABLE "search_terms" (
	"search_termID"	INTEGER,
	"symbolID"	INTEGER,
	"topicID"	INTEGER,
	"search_terms"	TEXT,
	"iteration"	INTEGER,
	PRIMARY KEY("search_termID")
);`

I had to remove your constraints on my system, but you can put them back.
Basically, change 'int' to 'integer' and 'varchar' to 'text'.

If you add a record, it'll now work.

![image](https://user-images.githubusercontent.com/7872218/100382708-cc98a600-3013-11eb-8eab-7a05a69695c1.png)

**Comment by jimbasquiat** [comment]
NICE thanks a lot!

**Comment by chrisjlocke** [comment]
For the avoidance of doubt, SQLite only knows about these types:

![image](https://user-images.githubusercontent.com/7872218/100382773-f5b93680-3013-11eb-8680-d877ee27722e.png)

These should be pretty obvious.  Store dates in text fields, as either yyyy-MM-dd HH:mm:ss (2020-11-26 18:20:19) or as integers (seconds since 1970).  I prefer the former as they're human readable.
If you ignore blobs (store images in there) just concentrate on integers, texts and reals.  Reals are floating point numbers (123.456, etc).

**Comment by jimbasquiat** [comment]
Thanks again, super useful

**Comment by chrisjlocke** [comment]
No problem.  Yell if you've any issues.  Primarily this is for support for DB4S, but we're happy to help in general SQL problems too. :)

**Comment by chrisjlocke** [comment]
I'll close this issue, but raise a new issue regarding the incorrect ticking of the PK option.

**Comment by chrisjlocke** [comment]
Actually, while recreating steps and writing it up, it does actually work out OK.

A table can have any field (numeric or textual) as a primary key.  It just has to be unique.  
In SQLite, if you set a primary key on a field marked as 'Integer', the 'auto-increment' is kind of implied - SQLite does this automatically.  That's not _strictly_ true - there is a big difference between primary key on an integer field and setting the 'auto-increment' option - mainly about re-using row numbers.  

However, if you set the 'AI' option, DB4S does make sure that's an integer.
Before:
![image](https://user-images.githubusercontent.com/7872218/100384323-cefcff00-3017-11eb-9f21-f875ca4d6d7a.png)

After:
![image](https://user-images.githubusercontent.com/7872218/100384332-d8866700-3017-11eb-92a1-70c164907a46.png)

So DB4S is sensible and helps out! ;)
For the main issue you experienced with no number being created - this is by design and isn't a bug, per see.

Yell back if ANY of that doesn't make sense...
